### ISSUE TITLE
Poisson solver diverges for some simple setup

### ISSUE THREAD

**Issue by RemiLehe** [ECP-WarpX/WarpX#5428]
The Poisson solver diverges for this simple input script:
```
max_step = 15
warpx.const_dt = 1e-09

geometry.dims = RZ
geometry.prob_hi = 0.03 0.1
geometry.prob_lo = 0 -0.1
amr.n_cell = 64 512
amr.max_grid_size = 64
amr.max_level = 0

boundary.field_hi = pec pec
boundary.field_lo = none pec
boundary.particle_hi = absorbing absorbing
boundary.particle_lo = none absorbing

warpx.do_electrostatic = labframe
warpx.eb_implicit_function = "-0.5 + (z>0)*(z<=0.01)*(x>0.01)*(x<=0.02)"
warpx.eb_potential(x,y,z,t) = "1"
warpx.self_fields_absolute_tolerance = 0.0001

diagnostics.diags_names = fields
fields.diag_type = Full
fields.format = openpmd
fields.intervals = 10
```
with the following message:
```
MLMG: # of AMR levels: 1
      # of MG levels on the coarsest AMR level: 5
MLMG: Initial rhs               = 0
MLMG: Initial residual (resid0) = 13107199.98
MLMG: Iteration   1 Fine resid/resid0 = 4.231904397e+31
MLMG: Failing to converge after 1 iterations. resid, resid/resid0 = 5.546841721e+38, 4.231904397e+31
amrex::Abort::0::MLMG failing so lets stop here !!!
```

WarpX was compiled with:
```
cmake -S . -B build -DWarpX_DIMS=RZ -DWarpX_EB=ON
cmake --build build -j 4
```
and run with:
```
./warpx.rz.MPI.OMP.DP.PDP.OPMD.EB.QED inputs
```

However, the Poisson solver converges if I apply either of the following changes:
- Add `eb2.small_volfrac = 1e-8` in the input script
- Change `(z>0)` to `(z>-1e-300)` in `warpx.eb_implicit_function = "-0.5 + (z>0)*(z<=0.01)*(x>0.01)*(x<=0.02)"`

**Comment by dpgrote** [comment]
I did some checking in Warp, looking at cases where the conductor lies exactly on or near grid cells. I don't see any problems arising, with the solver always converging in the same number of iterations and the solution being as expected. I used a test case that setups a box conductor and set its size so that the edge of the box is on or near grid cells. I included cases where the conductor edge is slightly on one side or the other of the cell (by 1.e-10*cell size).

I did the same test with WarpX, and for this example, there were no issues either.
